src/pages/BookDetailsPage/BookDetails/BookAction/BookSubscribeAction/index.tsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable react/jsx-no-useless-fragment */
+import "./BookSubscribeAction.css";
+import { Typography } from "@mui/material";
+import { useParams } from "react-router";
+import useSubscribe from "../../../../../hooks/useSubscribe";
+import { useSuccessScreenContext } from "../../../../../store/books/success/SuccessScreenContext";
+
+const successMessage = "You have successfully subscribed";
+const gratitudeMessage =
+  "You will receive a notification once the book becomes available";
+
+const BookSubscribeAction = () => {
+  const { bookId } = useParams();
+  const { isSubscribed, subscribe, unsubscribe, getSubscription } =
+    useSubscribe(parseInt(bookId || "0", 10));
+  const { showSuccessScreen } = useSuccessScreenContext();
+
+  const subscribeClick = () => {
+    subscribe().then(() => {
```

Comment:
What happens when subscribe or unsubscribe api fails? We should think of showing error message for the user.